kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/k8s/HelmClient.java

```diff
@@ -152,4 +162,38 @@ private String mapToValuesParameterArgument(Map<String, String> m) {
     private static String concatEntry(Map.Entry<String, String> entry) {
         return String.format("%s=%s", entry.getKey(), entry.getValue());
     }
+
+    /**
+     * Gets chart version.
+     *
+     * @param chartName the chart name
+     * @param releaseVersion the release version
+     * @return the chart versions
+     */
+    public String getChartVersion(String chartName, String releaseVersion) {
+        String chartVersion = "";
+        if (releaseVersion != null && !releaseVersion.isEmpty()) {
+            LOGGER.info("Getting chart version for {}", chartName);
+            ExecResult result = Exec.exec(null, command("search", "repo",
+                    chartName,
+                    "--versions",
+                    "--output=json"), 0, true);
+
+            ObjectMapper mapper = new ObjectMapper();
+            try {
+                ArrayNode rootNode = (ArrayNode) mapper.readTree(result.out());
+                // Looking for the latest version that match with releaseVersion. If set to 1.15, it will look for the latest 1.15.Z
+                for (JsonNode node : rootNode) {
+                    if (node.at("/app_version").asText().startsWith(releaseVersion)) {
+                        chartVersion = node.at("/version").asText();
+                        break;
+                    }
+                }
+            }
+            catch (JsonProcessingException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+        return chartVersion;
```

Comment:
Best abort if we don't find what we need.  Falling back to latest would mean we test against the wrong code.